File: problem_0047/problem_0047.c
#include <stdio.h>

int main(){

    int count = 0;

    int i, j;

    for (i = 1; i<1000000; i++){
        int temp = i;
        int divs = 0;

        if (temp % 2 == 0){
            divs++;
            while (temp % 2 == 0){
                temp /= 2;
            }
        }

        for (j = 3; j<1000; j+=2){
            if (temp % j == 0){
                divs++;
                while (temp % j == 0){
                    temp /= j;
                }
            }
        }

        if (divs == 4){
            count++;
        }
        else{
            count = 0;
        }

        if (count == 4){
            printf("%d\n", i-3);
            break;
        }
    }

    return 0;

}

// answer: 134043

File: python/problem_0047.py
count = 0 # number of consecutive numbers with 4 distinct prime factors

for i in range(1, 1000000):
    temp = i # temp variable to store i
    divs = 0 # number of distinct prime factors

    if i % 2 == 0:
        divs = 1
        while i % 2 == 0:
            i = int(i / 2)

    for j in range(3, 1000, 2):
        if i % j == 0:
            divs += 1
            while i % j == 0:
                i = int(i / j)
    if divs == 4:
        count += 1
    else:
        count = 0
    if count == 4:
        print(temp - 3)
        break

# answer: 134043
